blinkm/blinkm.go

```diff
@@ -0,0 +1,54 @@
+// Package blinkm implements a driver for the BlinkM I2C RGB LED.
+//
+// Datasheet: http://thingm.com/fileadmin/thingm/downloads/BlinkM_datasheet.pdf
+package blinkm
+
+import (
+	"machine"
+)
+
+// Device wraps an I2C connection to a BlinkM device.
+type Device struct {
+	bus machine.I2C
+}
+
+// New creates a new BlinkM connection. The I2C bus must already be
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus machine.I2C) Device {
+	return Device{bus}
+}
+
+// Version returns the version of firmware on the BlinkM.
+func (d Device) Version() ([]byte, error) {
+	version := []byte{0, 0}
+	d.bus.ReadRegister(Address, GET_FIRMWARE, version)
+	return version, nil
```

Comment:
This causes a heap allocation. You can avoid that by returning the two numbers itself (`return version[0], version[1], nil`) or as a single integer (`return uint16(version[0]) << 8 | uint16(version[1]), nil`).
Many libraries expose a version number as a single number, for example, FastLED is at version 3002001 at the moment (meaning 3.2.1). Such a version number is also easier to compare.

Reply:
Oddly the return values are actually letters example `a e` not numbers. But probably the same idea still applies:

```
func (d Device) Version() (major, minor byte, err error)
```

What do you think?

Reply:
Sounds good!

Reply:
Done.